packages/core/src/components/panel-stack/panelHeader.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+
+import * as React from "react";
+
+import { Icon } from "../icon/icon";
+import { Text } from "../text/text";
+
+import * as Classes from "../../common/classes";
+
+export interface IBackButton {
+    title: string;
+    onClick(): void;
+}
+
+export interface IPanelHeaderProps {
+    back?: IBackButton;
```

Comment:
prefer to avoid objects as props as it often breaks the pure behavior by creating new objects every render ([you do this below](#diff-e8ab461dbd0d083bb148b0cb65f9dac2R21)). spell them out as `backTitle` and `onBackClick`.